picoCTF-web/api/apps/v1/groups.py

```diff
@@ -384,3 +384,75 @@ def none_to_empty(self, in_data):
             'success': True,
             'accounts': created_accounts
         })
+
+
+@ns.route('/<string:group_id>/scoreboard')
+class ScoreboardPage(Resource):
+    """
+    Get a scoreboard page for a group.
+
+    If a page is not specified, will attempt to return the page containing the
+    current team, falling back to the first page if neccessary.
+    """
+    @block_before_competition
+    @ns.response(200, 'Success')
+    @ns.response(403, 'Permission denied')
+    @ns.response(404, 'Group not found')
+    @ns.response(422, 'Competition has not started')
+    @ns.expect(scoreboard_page_req)
+    def get(self, group_id):
+        """Retrieve a scoreboard page for a group."""
+        group = api.group.get_group(gid=group_id)
+        if not group:
+            raise PicoException('Group not found', 404)
+        group_members = [group['owner']] + group['members'] + group['teachers']
+
+        curr_user = api.user.get_user()
+        if (not curr_user or (curr_user['tid'] not in group_members
+                              and not curr_user['admin'])):
+            raise PicoException("You do not have permission to " +
+                                "view this group's scoreboard.", 403)
+
+        req = scoreboard_page_req.parse_args(strict=True)
+        if req['search'] is not None:
+            page = api.stats.get_filtered_scoreboard_page(
+                {'group_id': group_id},
+                req['search'], req['page'] or 1
+            )
+        else:
+            page = api.stats.get_scoreboard_page(
+                {'group_id': group_id}, req['page'])
+        return jsonify({
+            'scoreboard': page[0],
+            'current_page': page[1],
+            'total_pages': page[2]
+        })
+
+
+@ns.route('/<string:group_id>/score_progressions')
+class ScoreProgressionsResult(Resource):
+    """Get a list of score progressions for the top n teams in a group."""
+
+    @block_before_competition
+    @ns.response(200, 'Success')
+    @ns.response(403, 'Permission denied')
+    @ns.response(404, 'Group not found')
+    @ns.response(422, 'Competition has not started')
+    @ns.expect(score_progressions_req)
+    def get(self, group_id):
+        """Get a list of teams' score progressions."""
+        req = score_progressions_req.parse_args(strict=True)
+        group = api.group.get_group(gid=group_id)
+        if not group:
+            raise PicoException('Group not found', 404)
+        group_members = [group['owner']] + group['members'] + group['teachers']
+
+        curr_user = api.user.get_user()
+        if (not curr_user or (curr_user['tid'] not in group_members
+                              and not curr_user['admin'])):
+            raise PicoException("You do not have permission to " +
+                                "view this group's score progressions.", 403)
+        return jsonify(
+            api.stats.get_top_teams_score_progressions(
+                limit=req['limit'], group_id=group_id
```

Comment:
let's remove the limit kwarg here, and force it to default when called form here (maybe allow admin somewhere else later). otherwise it could get very taxing if someone inputs 60k or so. This will make it line up with the cache_stats daemon